src/main/java/org/asamk/signal/http/HttpServerHandler.java

```diff
@@ -0,0 +1,123 @@
+package org.asamk.signal.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpServer;
+
+import org.asamk.signal.commands.Commands;
+import org.asamk.signal.jsonrpc.JsonRpcReader;
+import org.asamk.signal.jsonrpc.JsonRpcResponse;
+import org.asamk.signal.jsonrpc.JsonRpcSender;
+import org.asamk.signal.jsonrpc.SignalJsonRpcCommandHandler;
+import org.asamk.signal.manager.Manager;
+import org.asamk.signal.manager.MultiAccountManager;
+import org.asamk.signal.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.concurrent.Executors;
+
+public class HttpServerHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(HttpServerHandler.class);
+
+    private final ObjectMapper objectMapper = Util.createJsonObjectMapper();
+
+    private final InetSocketAddress address;
+
+    private final Manager m;
+
+    private final MultiAccountManager c;
+
+    public HttpServerHandler(final InetSocketAddress address, final Manager m) {
+        this.address = address;
+        this.m = m;
+        this.c = null;
+    }
+
+    public HttpServerHandler(final InetSocketAddress address, final MultiAccountManager c) {
+        this.address = address;
+        this.m = null;
+        this.c = c;
+    }
+
+    public void init() {
+
+        try {
+
+            logger.info("Starting server on port " + address.toString());
+
+            final var server = HttpServer.create(new InetSocketAddress(address.getPort()), 0);
+            server.setExecutor(Executors.newFixedThreadPool(10));
+
+            server.createContext("/api/v1/rpc", httpExchange -> {
+
+                if (!"POST".equals(httpExchange.getRequestMethod())) {
+                    sendResponse(405, null, httpExchange);
+                    return;
+                }
+
+                try {
+
+                    final SignalJsonRpcCommandHandler commandHandler;
+
+                    if (c != null) {
+                        commandHandler = new SignalJsonRpcCommandHandler(c, Commands::getCommand);
+                    } else {
+                        commandHandler = new SignalJsonRpcCommandHandler(m, Commands::getCommand);
+                    }
```

Comment:
`c` and `m` don't change, so you could actually create the CommandHandler already in the constructor.

Reply:
Makes sense. I just wasn't sure about the thread safety of ithis, so I went the safe route. But if you thing that it will be safe then that is much better and efficient.

Reply:
Good point, yes the CommandHandler is thread safe, it doesn't have any mutable state and just uses the Managers, which are thread safe.